src/tracer.js

```diff
@@ -253,6 +273,38 @@ export default class Tracer {
     }
 
     /**
+     * Reapplies sampling if the parent span isn't sampled
+     *
+     * @param parent the parent span context
+     * @param ctx the span context
+     * @param operationName the operation name for the child
+     * @param tags tags to be applied by the sampler
+     */
+    handleUpsampling(parent: SpanContext, ctx: SpanContext, operationName: string, tags: any) {
+        if (!this._upsampling.enabled) {
+            return
+        }
+
+        // Run upsampler on every span
+        if (this._upsampling.allowOnInnerSpans) {
+            if (this._sampler.isSampled(operationName, tags)) {
+                ctx.samplingDecision = true;
+            }
+            return
+        }
+
+        // Reuse upsampling decision, if one exists in parent
+        if (parent.upsamplingDecision) {
```

Comment:
No, it cannot because if `allowOnInnerSpans` is set, we would use need a new sampling decision on every new span. We shouldn't reuse sampling decisions.